src/App/Program.fs

```diff
@@ -278,6 +279,12 @@ type MissionOptions
              Default = 0)>]
     member self.PubnetParallelCatchupStartingLedger = pubnetParallelCatchupStartingLedger
 
+    [<Option("in-memory-pubnet-parallel-catchup",
```

Comment:
Hmm, this feels a little not-orthogonal-enough / too special-cased. We already have an internal config flag (only shuffled around internally and settable from missions, not surfaced at the command line) called `inMemoryMode` which is present on `StellarCoreCfg` and `CoreSetOptions`. Is it possible to make this a general `--in-memory` flag that sets that option (or maybe sets a context option, that the existing 2 uses delegate to)?

Reply:
I'm not so sure about this. That would make the `--in-memory` flag configurable for the `InMemoryMode` mission, and that mission doesn't make sense to run without that flag. I named the new option `inMemoryPubnetParallelCatchup` (I can change this to `inMemoryParallelCatchup` so we can use it for testnet as well) specifically to distinguish it from the `inMemoryMode` option used in `StellarCoreSets` because parallel catchup doesn't use core sets. Also, many of the missions use core as a validator, so testing with captive core isn't useful in those cases.

I can tie the context option that is set on the command line in this PR to the `InMemoryMode` flag that is set on the core sets, but that's a somewhat significant change and  I'm don't think we would gain much by doing that, unless we think we'll toggle the `--in-memory` flag for other missions.